### PR TITLE
solution

### PR DESCRIPTION
**Pull request by kostya-konev**

### CODE REVIEW

File: src/convertToObject.js [hunk 1]
```diff
@@ -3,16 +3,31 @@
...
-  // write your code here
+  const cssObj = {};
+
+  const cssArray = sourceString.split(';');
+  const correctedArray = cssArray.map(elem => elem.trim());
+  const withoutEmpty = correctedArray.filter(elem => elem.length > 0);
+  const divided = withoutEmpty.map(elem => elem.split(':'));
+  const dividedWithoutSpaces = divided.map(elem => elem.map(
+    word => word.trim()));
+
```
Comment (vkryvytskyy) [inline src/convertToObject.js:25]: use chaining

File: src/convertToObject.js [hunk 2]
```diff
@@ -3,16 +3,31 @@
...
+  dividedWithoutSpaces.forEach((elem, i) => {
```
Comment (vkryvytskyy) [inline src/convertToObject.js:26]:
```suggestion
  dividedWithoutSpaces.forEach(([key, value], i) => {
```

**Review by vkryvytskyy — CHANGES_REQUESTED** [review]

File: src/convertToObject.js [hunk 3]
```diff
@@ -3,16 +3,28 @@
...
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cssObj = {};
+
+  const cssArray = sourceString.split(';').map(elem =>
+    elem.trim()).filter(elem => elem.length > 0).map(elem =>
+    elem.split(':')).map(elem => elem.map(word => word.trim()));
```
Comment (vkryvytskyy) [inline src/convertToObject.js:21]:
```suggestion
  const cssArray = sourceString
    .split(';')
    .map(elem => elem.trim())
    .filter(elem => elem.length > 0)
    .map(elem => elem.split(':'))
    .map(elem => elem.map(word => word.trim()));
```

File: src/convertToObject.js [hunk 4]
```diff
@@ -3,16 +3,28 @@
...
+
+  cssArray.forEach(([key, value], i) => {
```
Comment (vkryvytskyy) [inline src/convertToObject.js:23]:
```suggestion
  cssArray.forEach(([key, value]) => {
```
`i` is not used anymore

**Review by vkryvytskyy — CHANGES_REQUESTED** [review]

**Review by marina-tilniak — APPROVED** [review]

**Pull request closed without merging**